core/src/main/java/org/apache/calcite/adapter/enumerable/RexToLixTranslator.java

```diff
@@ -312,7 +312,9 @@ private Expression getConvertExpression(
       case CHAR:
       case VARCHAR:
         return Expressions.call(BuiltInMethod.ST_GEOM_FROM_EWKT.method, operand);
-
+      case BINARY:
+      case VARBINARY:
+        return Expressions.call(BuiltInMethod.ST_GEOM_FROM_EWKB.method, operand);
```

Comment:
As a future consideration, there are other binary formats like Tiny WKB.  https://github.com/TWKB/Specification/blob/master/twkb.md

I suppose that may be handled by a specific cast.

Reply:
Thanks a lot for the pointer, it would definitely make sense to implement these functions:
https://postgis.net/docs/ST_AsTWKB.html